cosmos/lib/cosmos/microservices/interface_microservice.rb

```diff
@@ -330,9 +330,8 @@ def run
                   handle_packet(packet)
                   @count += 1
                 else
-                  Logger.info "#{@interface.name}: Clean disconnect (returned nil)"
-                  # Don't reconnect on a clean disconnect because it's intentional
-                  handle_connection_lost(reconnect: false)
```

Comment:
Hmm, we seemed to be doing this explicitly. Is this a change from 4 to 5? When you click Disconnect does that kill the connection thread so this code doesn't matter?

Reply:
I haven't verified that this doesn't break the disconnect button, but it shouldn't.  Like you said, that should kill the read thread.  I'll check that before merging.   This is back to the behavior of COSMOS 4, and what I think is the correct behavior.  Even if the target closes the socket, COSMOS should still try to reconnect. That would happen whenever a target does a clean reboot.